open-craft/opencraft
# [BB-6370] feat: add instance configuration settings to grove payload

## Description

This PR adds support for including the following configuration settings to the grove trigger payload, while triggering a new deployment:

1. Theme repo information and theme override settings
2. Site configuration settings
3. LMS and CMS env configuration settings

## Supporting information

[BB-6370](https://tasks.opencraft.com/browse/BB-6370)

## Testing instructions

1. Checkout this branch
2. Add the grove related configuration to the .env.test file
3. Setup local environment using [these instructions](https://ocim.opencraft.com/en/latest/development/docker/)
4. From the django shell, create a new instance using [these instructions](https://ocim.opencraft.com/en/latest/provisioning-sandboxes/)
5. Go to [console ui](http://localhost:3000/) and signup. This should create a new BetaTestApplication.
6. Go to the [django admin page](http://localhost:5000/admin/registration/betatestapplication/), and add the instance id of the Grove instance and instance type as `grove instance` for the newly create BetaTestApplication
7. Go to your Grove instance in the [django admin panel](http://localhost:5000/admin/grove/groveinstance/).
8. Enable `Deploy simple_theme` checkbox
9. Add sample theme settings such as this :
```json
{"version": 1, "footer-bg": "#FFFFFF", "header-bg": "#FFFFFF", "link-color": "#126F9A", "main-color": "#126F9A", "btn-primary-bg": "#126F9A", "btn-secondary-bg": "#FFFFFF", "btn-primary-color": "#FFFFFF", "btn-secondary-color": "#126F9A", "btn-primary-hover-bg": "#FFFFFF", "btn-secondary-hover-bg": "#126F9A", "btn-primary-hover-color": "#126F9A", "btn-primary-border-color": "#126F9A", "btn-secondary-hover-color": "#FFFFFF", "btn-secondary-border-color": "#126F9A", "home-page-hero-title-color": "#313131", "home-page-hero-subtitle-color": "#646464", "btn-primary-hover-border-color": "#126F9A", "btn-secondary-hover-border-color": "#FFFFFF"}
```
10. Add sample static content in the `Static content overrides JSON` field as this one:
```json
{"version": 0, "homepage_overlay_html": "<h1>Welcome to Kaustav Testing</h1><p>It works! Powered by Open edX\u00ae</p>"}
```
11. Add sample env configuration in the `Configuration extra settings` field such as this one:
```yaml
EDXAPP_LMS_ENV: 
    DCS_SESSION_COOKIE_SAMESITE: Lax
EDXAPP_LMS_ENV_FEATURES:
    ENABLE_CREATOR_GROUP: true
EDXAPP_CMS_ENV:
    SESSION_ENGINE: django.contrib.sessions.backends.cached_db
EDXAPP_CMS_ENV_FEATURES:
    ENABLE_BULK_ENROLLMENT_VIEW: true
EDXAPP_COMMON_ENV_FEATURES:
    ENABLE_COUNTRY_ACCESS: true
EDXAPP_LOG_LEVEL: info
EDXAPP_PDF_RECEIPT_TAX_ID: TAX123
EDXAPP_CMS_STATIC_URL_BASE: http://static.example.com
EDXAPP_ENABLE_MKTG_SITE: true
```
12. From the [console backend UI](http://localhost:5000/instance/), trigger a new deployment
13. Go the the gitlab pipeline for the configured Grove repo, and verify the payload as well as the resulting configuration of the instances.


## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
